src/Tools/ILAsm/IlAsmDeploy.csproj

```diff
@@ -46,7 +46,7 @@
   </PropertyGroup>
 
   <Target Name="_PublishILAsm"
-          AfterTargets="Build"
+          BeforeTargets="AfterBuild"
```

Comment:
Can you elaborate on how this fixes things?

Reply:
I tried to reproduce in a small solution but was unable to.

@rainersigwald Any idea why would there be a difference between running msbuild from command line and from VS re execution of targets that are chained after `Build` target?

Reply:
There's a weird conceptual hole here that MSBuild doesn't behave super consistently around.

The problem is this: if you build target `Foo` in a project, and there's no dependency chain that reaches from `Foo` to `Bar`, you probably don't care if `Bar` fails. It may be relevant later, but not when you say `Build(project: whatever, target: Foo)`.

That makes sense. Where things get complicated is AfterTargets. If `AfterFoo AfterTargets="Foo"` , then whenever `Foo` runs, `AfterFoo` will run. But what does that mean if there was a build request for `target: Foo`? Should it succeed, because `Foo` did, or fail, because `AfterFoo` did? For that matter, _when should it return?_ When the part you asked for is done? Or when all the stuff that happened as a consequence of what you asked for is done?

MSBuild doesn't behave super consistently around these questions. For example, we fail build requests where `AfterFoo` fails . . . but only in single-proc builds (https://github.com/Microsoft/msbuild/issues/3345).

Because of this weirdness, I like this change: it moves the custom target earlier in the dependency chain, so it sidesteps this conceptual question and all the bugs that surround it.